### PR TITLE
[DRAF][one-optimize] Optimize part of the transformer's attention-head

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This draft introduces two new passes to optimize part of the transformer's attention-head: FuseMulWithFullyConnected and FuseStridedSlicesNegAsMul.

for issue:  #12917

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
> compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp:77:21: error: unused variable ‘fc_input’ [-Werror=unused-variable]

**Comment by seanshpark** [comment]
some comments;
- please add some test models in `res/TensorFlowLiteRecipes`
- and add to `circle2circle-dredd-recipe-test`
- also to `luci-pass-value-py-test`

**Comment by BalyshevArtem** [comment]
@seanshpark, can I split this draft into PRs?

**Comment by seanshpark** [comment]
> can I split this draft into PRs?

Yes, please do :)
Actually I don't do detailed reviewed with draft PRs.

File: res/TensorFlowLiteRecipes/Net_StridedSlices_Neg_000/test.recipe [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  shape { dim: 1 dim: 1 dim: 4 }
+}
+operation {
+  type: "Concatenation"
+  concatenation_options {
+    axis: -1
+    activation: NONE
+  }
+  input: "output_1"
+  input: "output_neg"
```
Comment (jinevening) [inline res/TensorFlowLiteRecipes/Net_StridedSlices_Neg_000/test.recipe:98]: In the target model, the inputs are swapped. PTAL https://github.com/Samsung/ONE/issues/12917#issuecomment-2099624359

**Pull request closed without merging**